drivers/hmis/app/models/hmis/hud/custom_case_note.rb

```diff
@@ -0,0 +1,24 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Hmis::Hud::CustomCaseNote < Hmis::Hud::Base
+  self.table_name = :CustomCaseNote
+  self.sequence_name = "public.\"#{table_name}_id_seq\""
+
+  belongs_to :client, **hmis_relation(:PersonalID, 'Client')
+  belongs_to :enrollment, **hmis_relation(:EnrollmentID, 'Enrollment')
+  belongs_to :user, **hmis_relation(:UserID, 'User')
+  belongs_to :data_source, class_name: 'GrdaWarehouse::DataSource'
+  alias_to_underscore [:CustomCaseNoteID, :PersonalID, :EnrollmentID]
+
+  before_create do |record|
+    record.CustomCaseNoteID ||= Hmis::Hud::Base.generate_uuid
+  end
+
+  replace_scope :viewable_by, ->(user) do
+    joins(:client).merge(Hmis::Hud::Client.viewable_by(user))
```

Comment:
This needs to be a bit more complicated. If the note is tied to an Enrollment, the user should only be allowed to see the note if they are allowed to see the enrollment.

It looks like that's maybe not in place for Files, but it really should be.

Reply:
got it, should I change it for files too in this PR?

Reply:
seemed like the file perm change was causing some failures. [Moved it to another PR](https://github.com/greenriver/hmis-warehouse/pull/3524) so this doesn't get blocked